cluster/join.sls

```diff
@@ -14,6 +16,15 @@ wait-for-total-initialization:
     - require:
       - wait-for-cluster
 
+{% if cluster.join_lock %}
+acquire-lock-cluster:
+  cmd.run:
+    - name: ssh -T root@{{ cluster.init }} -o StrictHostKeyChecking=no '[ ! -f "{{ lock_file }}" ] && touch {{ lock_file }}'
+    - retry:
+        attempts: 30
```

Comment:
regarding the timout we still have hardcoded it in many places, so I think we don't need to make a parametrizable timeout here.